### PR TITLE
NIFI-6598 Storing peers into managed-state

### PR DESCRIPTION
**Pull request by ijokarumawak**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

- Write peers into managed-state, instead of local file
- Now, the local file location constants and its property key are no longer needed
- Confirmed that state is written
- Implemented recovering peers from the stored state
- Tested storing/recovering using a local file
- Tested with a cluster where the bootstrap node (specified at the RPG) is not reachable, but the one of stored peer is still alive. In that case, NiFi will restore remote peer information from managed state when it restarted, then RPG should be able to transfer FlowFiles with the remaining nodes. Confirmed this scenario works as expected.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/RemoteProcessGroupResource.java [hunk 1]
```diff
@@ -827,6 +829,61 @@ public Response updateRemoteProcessGroupRunStatus(
...
+        @PathParam("id") final String id) throws InterruptedException {
+
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.GET);
+        }
+
+        // authorize access
+        serviceFacade.authorizeAccess(lookup -> {
+            final Authorizable authorizable = lookup.getRemoteProcessGroup(id);
+            authorizable.authorize(authorizer, RequestAction.WRITE, NiFiUserUtils.getNiFiUser());
```
[the remark this answers is not in the document]
Comment (ijokarumawak) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/RemoteProcessGroupResource.java]: Yes, I thought, so. However, other existing endpoints such as Processor and Controller state require WRITE privilege. So I followed that.
https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessorResource.java#L404

File: nifi-commons/nifi-site-to-site-client/src/main/java/org/apache/nifi/remote/client/SiteToSiteClient.java [hunk 2]
```diff
@@ -921,6 +931,23 @@ public File getPeerPersistenceFile() {
...
+            if (stateManager != null) {
+                return new StatePeerPersistence(stateManager);
+
+            } else if (peerPersistenceFile != null) {
+                return new FilePeerPersistence(peerPersistenceFile);
+            }
+
+            return null;
+        }
+
```
[the remark this answers is not in the document]
Comment (ijokarumawak) [reply nifi-commons/nifi-site-to-site-client/src/main/java/org/apache/nifi/remote/client/SiteToSiteClient.java]: Great catch! I will consider how we can make it work in such context. IMHO, if serialization works if we mark the StateManager in the config as `transient`, then that'd be the least impact approach. I will test that.

**Comment by ijokarumawak** [comment]
@bbende I've made the stateManager transient and added more detailed docs for the use of persistent file and stateManager. I hope this suffice. Also, rebased with the latest master to fix merge conflict. Please have another look. Thanks!

**Comment by bbende** [comment]
Looks good, going to merge, thanks!

**Pull request closed without merging**